recipes/libsodium/1.0.18/conanfile.py

```diff
@@ -0,0 +1,223 @@
+        self._build_autotools_impl(configure_args)
+
+    def _build_autotools_mingw(self, configure_args):
+        if self.settings.arch == "x86":
+            arch = "i686"
+        else:
+            arch = "x86_64"
+        host_arch = "%s-w64-mingw32" % arch
+        configure_args.append("--host=%s" % host_arch)
+        self._build_autotools_impl(configure_args)
+
+    def _build_autotools_darwin(self, configure_args):
+        if self.settings.os == "iOS":
+            os = "ios"
+        else:
+            os = "darwin"
+        host_arch = "%s-apple-%s" % (self.settings.arch, os)
+        configure_args.append("--host=%s" % host_arch)
+        self._build_autotools_impl(configure_args)
+
+    def _build_autotools(self):
+        absolute_install_dir = os.path.abspath(os.path.join(".", "install"))
+        absolute_install_dir = absolute_install_dir.replace("\\", "/")
+        configure_args = self._get_configure_args(absolute_install_dir)
+
+        if self.settings.os == "Linux":
+            self._build_autotools_linux(configure_args)
+        elif self.settings.os == "Emscripten":
+            self._build_autotools_emscripten(configure_args)
+        elif self.settings.os == "Android":
+            self._build_autotools_android(configure_args)
+        elif self.settings.os in ["Macos", "iOS", "watchOS", "tvOS"]:
+            self._build_autotools_darwin(configure_args)
+        elif self.settings.os == "Windows" and self.settings.compiler == "gcc":
+            self._build_autotools_mingw(configure_args)
+        else:
+            raise ConanInvalidConfiguration(f"Unsupported os for libsodium: {self.settings.os}")
+
+    def build(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+        if self.settings.os == "Macos":
+            tools.replace_in_file(os.path.join(self._source_subfolder, "configure"), r"-install_name \$rpath/", "-install_name ")
+        if self.settings.compiler != "Visual Studio":
+            self._build_autotools()
+        else:
+            self._build_visual()
+
+    def package(self):
+        self.copy("*LICENSE", dst="licenses", keep_path=False)
+        if self.settings.compiler == "Visual Studio":
+            self._package_visual()
+        else:
+            self._package_autotools()
+
+    def package_info(self):
+        if self.settings.compiler == "Visual Studio":
+            if not self.options.shared:
+                self.cpp_info.defines = ["SODIUM_STATIC=1"]
+        self.cpp_info.libs = tools.collect_libs(self)
+        if self.settings.os == "Linux":
+            self.cpp_info.system_libs = ["pthread"]
+
+    def _package_autotools(self):
+        if self.settings.os == "Emscripten":
+            prefix = "%s/libsodium-js" % self._source_subfolder
+        else:
+            prefix = "install"
+        lib_folder = os.path.join(prefix, "lib")
+        self.copy("*.h", dst="include", src=os.path.join(prefix, "include"))
```

Comment:
do we need manual copies if we did `make install`?

Reply:
libsodium installs both static and shared libraries, plus it installs pkgconfig files

Reply:
Also, the build process is totally different between autotools and msbuild